build.sbt

```diff
@@ -134,7 +137,10 @@ lazy val coreTestMacros = crossProject(JSPlatform, JVMPlatform, NativePlatform)
 lazy val core = crossProject(JSPlatform, JVMPlatform, NativePlatform)
   .crossType(CrossType.Full)
   .configureCross(configureJUnit)
-  .settings(moduleName := "shapeless")
+  .settings(
+    moduleName := "shapeless",
+    javaModuleName := "shapeless.core"
```

Comment:
Perhaps we should use just `shapeless` for this one? Shapeless 2 is not really modular, the plugin and scratch are not published artifacts.